src/main/java/org/reaktivity/nukleus/tcp/internal/writer/stream/StreamFactory.java

```diff
@@ -159,10 +179,11 @@ private void processEnd(
             int offset,
             int limit)
         {
-            endRO.wrap(buffer, offset, limit);
-
-            // TODO: flush partial writes first (if necessary)
-            doCleanup();
+            if (slot == NO_SLOT) // no partial writes pending
+            {
+                endRO.wrap(buffer, offset, limit);
+                doCleanup();
+            }
```

Comment:
Added test case for DATA after END with deferred cleanup and made the suggested fix. NOTE: as a result of the fix we will immediately reset when window exceeded. We are not ddeferring the reset in order to honor the deferred write and END of stream. Doing so would make the code more complex and seems unnecessary since window exceeded is a case that would only arise because of a code bug in the target nukleus.